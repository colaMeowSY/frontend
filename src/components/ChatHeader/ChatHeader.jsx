import React from 'react';
import { IconButton } from '@mui/material';
import PropTypes from 'prop-types';
import CallIcon from '@mui/icons-material/Call';
import ProfileBadge from '../ProfileBadge/ProfileBadge';
import styles from './ChatHeader.module.css';

/**
 * ChatHeader component displays the avatar, name and a status of the buddy the current
 * user in the current conversation.
 * A call button will allow user to give a call to this buddy.
 */
function ChatHeader({ active, name, callBuddy }) {
  return (
    <div className={styles.chatHeader}>
      <div className={styles.userDetails}>
        <ProfileBadge active={active} name={name} />
        <div className={styles.detailsText}>
          <p
            className={`${styles.headerText} ${styles.buddyName} ${
              active ? styles.onlinePadding : styles.offlinePadding
            }`}>
            {name}
          </p>
          <p className={`${styles.headerText} ${styles.status}`}>{active ? 'Active now' : ' '}</p>
        </div>
      </div>
      <IconButton onClick={callBuddy} data-testid="call-buddy">
        <CallIcon fontSize="large" />
      </IconButton>
    </div>
  );
}

ChatHeader.propTypes = {
  active: PropTypes.bool,
  name: PropTypes.string.isRequired,
  callBuddy: PropTypes.func.isRequired,
};

ChatHeader.defaultProps = {
  active: false,
};

export default ChatHeader;
