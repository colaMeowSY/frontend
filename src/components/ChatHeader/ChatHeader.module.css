.chatHeader {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 10px 10px;
  background: white;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25), 1px 0px 0px #e8e8e8;
}

.userDetails {
  display: flex;
  align-items: center;
}

.headerText {
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 10px;
  font-family: 'Inter';
  font-style: normal;

}

.buddyName {
  font-weight: 600;
  font-size: 20px;
  line-height: 28px;
}

.onlinePadding {
  padding-top: 6px;
}

.offlinePadding {
  padding-top: 12px;
}

.detailsText {
  height: 52px;
}

.status {
  color: #959595;
  font-weight: 500;
  font-size: 14px;
}
