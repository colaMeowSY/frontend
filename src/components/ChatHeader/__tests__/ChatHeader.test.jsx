import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@mui/material';
import theme from '../../Theme';

import ChatHeader from '../ChatHeader';

describe('TestChatHeader', () => {
  test('should display buddy name', () => {
    render(
      
      <ThemeProvider theme={theme}>
        <ChatHeader
            name="Amy" callBuddy={() => {}}
        />
      </ThemeProvider>,
    );

    expect(screen.getByText('Amy')).toBeInTheDocument();
  });

  test('should display active text', () => {
    render(
      <ThemeProvider theme={theme}>
        <ChatHeader
          active
          name="Amy" callBuddy={() => {}}
        />
      </ThemeProvider>,
    );

    expect(screen.getByText('Active now')).toBeInTheDocument();
  });

  test('should call buddy when pressed', () => {
    const mockCallBuddy = jest.fn();

    render(
      <ThemeProvider theme={theme}>
        <ChatHeader
          name="Amy"
          callBuddy={mockCallBuddy}
        />
      </ThemeProvider>,
    );

    fireEvent.click(screen.getByTestId('call-buddy'));
    expect(mockCallBuddy).toHaveBeenCalledTimes(1);
  });
});

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ChatHeader name="Amy" callBuddy={() => {}}/>
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
