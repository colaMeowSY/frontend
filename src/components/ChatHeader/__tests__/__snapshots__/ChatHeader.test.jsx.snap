// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="chatHeader"
>
  <div
    className="userDetails"
  >
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-6ch1n3-MuiContainer-root"
    >
      <span
        className="MuiBadge-root css-wd6a4w-MuiBadge-root"
        data-testid="offline"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1pl8iwh-MuiAvatar-root"
          data-testid="Amy"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <span
          className="MuiBadge-anchorOriginBottomRightCircular MuiBadge-overlapCircular MuiBadge-badge MuiBadge-dot MuiBadge-anchorOriginBottomRight css-ank3r8-MuiBadge-badge"
        >
          
        </span>
      </span>
    </div>
    <div
      className="detailsText"
    >
      <p
        className="headerText buddyName offlinePadding"
      >
        Amy
      </p>
      <p
        className="headerText status"
      >
         
      </p>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
    data-testid="call-buddy"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
      data-testid="CallIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56-.35-.12-.74-.03-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99z"
      />
    </svg>
  </button>
</div>
`;
